{jd-gui,jd-cli}: drop

tomodachi94 opened the pull request:
Upstream has no commits since 2019:
https://github.com/java-decompiler/jd-gui

jd-gui uses Gradle 6, which has been EOL since 10 Feb 2023:
https://endoflife.date/gradle
(The informal "JDK team" is working on dropping Gradle 6).

Attempting to build with a newer Gradle fails with a nontrivial error.

Potential replacements: cfr, bytecode-viewer, procyon, jd-cli

Keeping this as a draft while @emilazy attempts to get it working.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

tomodachi94 committed:
jd-gui: drop

Upstream has no commits since 2019:
https://github.com/java-decompiler/jd-gui

jd-gui uses Gradle 6, which has been EOL since 10 Feb 2023:
https://endoflife.date/gradle
(The JDK team is working on dropping Gradle 6).

Attempting to build with a newer Gradle fails with a nontrivial error.

Potential replacements: cfr, bytecode-viewer, procyon

xtrm-en commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -566,6 +566,10 @@
 - The `services.patroni.raft` option has been removed, as Raft has been [deprecated by upstream since 3.0.0](https://github.com/patroni/patroni/blob/master/docs/releases.rst#version-300)
 
+- The `jd-gui` package was removed due to an inactive upstream and a dependency on the end-of-life Gradle 6.
+  Java decompilers already packaged in Nixpkgs include `jd-cli`, `cfr`,
```
Only `bytecode-viewer` looks like it's a complete alternative, providing a GUI, others seem to only provide a cli interface, maybe this can be specified?

tomodachi94 committed:
jd-cli: drop

JCenter Bintray is dead. This package depends on jd-core, which was
only hosted through JCenter Bintray.

tomodachi94 commented:
Updated to drop the jd-cli package　as suggested in `#jdk:nixos.org` by @emilazy, and address feedback by @xtrm-en.

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -566,6 +566,12 @@
 
 - The `services.patroni.raft` option has been removed, as Raft has been [deprecated by upstream since 3.0.0](https://github.com/patroni/patroni/blob/master/docs/releases.rst#version-300)
 
+- The `jd-cli` package was removed due to an inactive upstream and a dependency on the end-of-life Gradle 6.
```
No Gradle dependency here. It’s the dependency on the JCenter repository that was shut down.

tomodachi94 replied:
Oops, fixed.

emilazy commented on pkgs/top-level/aliases.nix:
```diff
@@ -533,6 +533,8 @@ mapAliases {
   jami-client-qt = jami-client; # Added 2022-11-06
   jami-client = jami; # Added 2023-02-10
   jami-daemon = jami.daemon; # Added 2023-02-10
+  jd-cli = throw "jd-cli has been removed due to upstream being unmaintained since 2019. Other Java decompilers in Nixpkgs include bytecode-viewer (GUI) cfr (CLI), and procyon (CLI)."; # Added 2024-10-30
+  jd-gui = throw "jd-gui has been removed due to upstream being unmaintained since 2019. Other Java decompilers in Nixpkgs include bytecode-viewer (GUI) cfr (CLI), and procyon (CLI)."; # Added 2024-10-30
```
Missing comma.

tomodachi94 replied:
Good catch.
